crates/wascc-provider/src/states.rs

```diff
@@ -8,3 +8,28 @@ pub(crate) mod running;
 pub(crate) mod starting;
 pub(crate) mod terminated;
 pub(crate) mod volume_mount;
+
+/// When called in a state's `next` function, exits the current state
+/// and transitions to the Error state.
+#[macro_export]
+macro_rules! transition_to_error {
+    ($slf:ident, $err:ident) => {{
+        let aerr = anyhow::Error::from($err);
+        log::error!("{:?}", aerr);
+        let error_state = super::error::Error {
+            message: aerr.to_string(),
+        };
+        return Transition::next($slf, error_state);
```

Comment:
Curiosity question here: Can we not just do `return Transition::next(self, error_state)`? This should only be used inside of a state function, so it should know what `self` is already, right?

Reply:
Unfortunately not, because Rust macros are hygienic - identifiers in the macro body are don't "join up" with identifiers around the calling site.

https://stackoverflow.com/questions/44120455/how-to-call-methods-on-self-in-macros

https://stackoverflow.com/questions/40593165/unable-to-use-self-in-macro-because-the-macro-expansion-ignores-token-self

https://users.rust-lang.org/t/how-to-use-self-or-any-variable-in-macro-generated-function-body/6264

Reply:
I had the same question haha.